struct_arm64.go

```diff
@@ -6,8 +6,58 @@ package purego
 import (
 	"math"
 	"reflect"
+	"unsafe"
 )
 
+func getStruct(outType reflect.Type, syscall syscall15Args) (v reflect.Value) {
+	outSize := outType.Size()
+	switch {
+	case outSize == 0:
+		return reflect.New(outType).Elem()
+	case outSize <= 8:
+		r1 := syscall.a1
+		if isAllSameFloat(outType) {
+			r1 = syscall.f1
+			if outType.NumField() == 2 {
+				r1 = syscall.f2<<32 | syscall.f1
+			}
+		}
+		return reflect.NewAt(outType, unsafe.Pointer(&struct{ a uintptr }{r1})).Elem()
+	case outSize <= 16:
+		r1, r2 := syscall.a1, syscall.a2
+		if isAllSameFloat(outType) {
+			switch outType.NumField() {
+			case 4:
+				r1 = syscall.f2<<32 | syscall.f1
+				r2 = syscall.f4<<32 | syscall.f3
+			case 3:
+				r1 = syscall.f2<<32 | syscall.f1
+				r2 = syscall.f3
+			case 2:
+				r1 = syscall.f1
+				r2 = syscall.f2
```

Comment:
That's so weird. So for two floats instead of cramming them into one reg, arm abi uses both regs?

Reply:
Oh I get it now, it's because they are more than 8 bytes in size, therefore double + float